### PR TITLE
Make generated cursor size configurable

### PR DESCRIPTION
**Pull request by DebauchedSloth**
Thanks so much for this project!  I've been looking for high quality cursors I could easily resize for HiDPI using LXQT, and this was a fantastic base.

There are a few changes in here:

(1) Make the size of the cursors configurable.  This includes both the conversion from svg, as well as the .cursor files.  I've removed the .cursor files and added .template files, and then I've converted the hotspots to be based off of nominal 100 (<small_size>) and 200 (<large_size>) cursors.  At runtime, the actual cursor files are generated using these template files and the selected cursor size, using a new chunk of python, conv.py.

(2) I've changed from inkscape to rsvg-convert.  rsvg-convert is vastly faster.

(3) I've changed the svg file iteration to only call sh once and to simply iterate over the files directly instead of using find.

(4) I changed build.sh to install automatically, and changed the install command to just use rsync.

I currently have the sizes set to 80/100; you should probably change that to your preferred size.

### CODE REVIEW

File: src/build.sh [hunk 1]
```diff
@@ -14,9 +18,8 @@ if [ ! -d "x2" ]; then
 fi
 
 cd svg/
-find . -name "*.svg" -type f -exec sh -c 'inkscape -z -e "../x1/${0%.svg}.png" -w 32 -h 32 $0' {} \;
-find . -name "*.svg" -type f -exec sh -c 'inkscape -z -e "../x2/${0%.svg}.png" -w 64 -w 64 $0' {} \;
-
+sh -c 'for f in *.svg ; do echo $f; rsvg-convert -o "../x1/$(echo $f | sed s/.svg/.png/)" -w ${SMALL_SIZE} -h ${SMALL_SIZE} "$f"; done'
+sh -c 'for f in *.svg ; do echo $f; rsvg-convert -o "../x2/$(echo $f | sed s/.svg/.png/)" -w ${LARGE_SIZE} -h ${LARGE_SIZE} "$f"; done'
```
Comment (keeferrourke) [inline src/build.sh]: same issue as above line.

File: src/config/conv.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+files = [join(source_path, f) for f in listdir(source_path) if isfile(join(source_path, f))]
+for fname in files:
+    if not fname.endswith(".template"):
+        continue
+    print("%s -> %s [%d %d]" % (fname, fname.replace(".template", ".cursor"), small_size, large_size))
+    f = open(fname, "r")
+    out = open(fname.replace(".template", ".cursor"), "w")
+    for line in f.readlines():
+        try:
```
Comment (keeferrourke) [inline src/config/conv.py]: In general I do not like try/exceptions, I consider this to be very bad style. This is especially demonstrated on line 25 - if you encounter an error while parsing these files you absolutely should not continue trying to do so.

**Comment by DebauchedSloth** [comment]
Thanks for the comments, and for the project.

**Comment by keeferrourke** [comment]
Hi there! Thanks for the contribution! There are just a few things that I need you to fix before I merge. Mainly the concerns are with your modifications to `build.sh`, but I will address them as you have listed your modifications.

> (1) Make the size of the cursors configurable. This includes both the conversion from svg, as well as the .cursor files. I've removed the .cursor files and added .template files, and then I've converted the hotspots to be based off of nominal 100 (<small_size>) and 200 (<large_size>) cursors.

Perhaps I am failing to understand this, but if the hotspots are fixed based on nominal sizes, and the height/width are configurable, then resizing them will make the hotspots wrong, thus rendering the built cursors difficult to use. The hotspots should be relative to the height/width if they are to be configurable.

> At runtime, the actual cursor files are generated using these template files and the selected cursor size, using a new chunk of python,`conv.py`.

This is all fine and good, as long as it makes sense, see above. Additionally, I have a disagreement with your use of try/except in `conv.py`, see the line review.

> (2) I've changed from `inkscape` to `rsvg-convert`. `rsvg-convert` is vastly faster.

Thanks for that! I wasn't aware that `rsvg-convert` existed, so the improvement in performance is welcome. I also really disliked including such a monstrous build dependency as `inkscape`.

I have concerns with your invocation of `rsvg-convert` however, see my line comments in `build.sh`. 

> (3) I've changed the svg file iteration to only call sh once and to simply iterate over the files directly instead of using find.

As above, I have concerns with this. See my line comments in `build.sh`. Also note that `build.sh` is a bash script, so I am not sure why you are invoking sh for a `for` loop. Just use the in-built functionality in bash.

> (4) I changed build.sh to install automatically, and changed the install command to just use `rsync`.

This is a problem. Build scripts are not installation scripts, and a user may or may not want to actually install the software, _let alone to the system as root_. 

Additionally, can you justify why you changed `cp` to `rsync`? Capitaine cursors is not large, and if the copying is interrupted, it is not integral that it be resumed intelligently later. One can simply overwrite the directory at no cost. You also did not change the recommended install method from `cp` to `rsync` for regular user installation and this inconsistency bothers me.

> I currently have the sizes set to 80/100; you should probably change that to your preferred size.

Re: the first point, if you have explicitly written the hotspots in the template files to work for a given height/width, then the hotspots must also change relative to the height/width in order to be useful.

**Comment by keeferrourke** [comment]
@DebauchedSloth thanks for your interest in my project :)

If you fix the issues I've outlined above, and reopen the request, I'll be more than happy to pull it into master.

Cheers mate!

**Pull request closed without merging**